### DISCUSSION TITLE
How to perform a proximity search with at least two words

### DISCUSSION THREAD

**Discussion by ishanuda** [krisk/Fuse#606] [Q&A] [answered]
Hi,
Is it possible to do a proximity search using the library.

EX: **foo 5 bar** (Where **foo** and **bar** should be occurred in a long sentence within the given distance of 5 words)

I went through the documentation but could not find any guideline or example.

Can any one provide an example or a reference to get an idea if possible?

Proximity search ref: https://en.wikipedia.org/wiki/Proximity_search_(text)

Thank you very much.

**Comment by krisk** [comment, marked as answer]
No, the library cannot do that on its own 😄 

Seems like a pretty neat use case though, and worth exploring more.

**Reply by ishanuda** [reply]
@krisk 
Thank you for the answer.

Can we request it as a feature?

**Reply by krisk** [reply]
I [posted](https://twitter.com/kirorisk/status/1522424482819809280?s=21&t=s8zZt-Hn_jT4II7Cdqf8og) about this on Twitter to get feedback as well, as this is considerable amount of work, so it would be good to get a rough understanding of how desirable this feature is 😄

**Reply by ishanuda** [reply]
Hi, @krisk,
I think, as a complete search library this is also a great feature that most of the end-users will be happy to experience in the same library.
As I can see this is a very powerful library.

Thank you very much for your reply.

**Comment by krisk** [comment]
I am not sure if this would be a feature used by "most" users though 😅 . Proximity-search is more of an advanced feature. Want to be mindful of also not introducing bloat to the library. One way to address this is by introducing plugin functionality, and developers can pick and choose which plugins to add to Fuse. 

I've been thinking of this plugin functionality for a long while, unfortunately, it is quite a lot of work.

**Comment by ishanuda** [comment]
Hi @krisk ,

Thank you for the reply.
I think that is a great idea to give an interface for the main library having extensibility.
So that the developers can advance the library with different kinds of features with an extension and end-users can use the extension if they needed the required functionality.
The main library can continue with the features that are most useful and needed as the basic functionality.
regards.
